### PR TITLE
Feature/saving by directory management

### PR DESCRIPTION
**Pull request by faildeny**
This PR addresses the problem of saving large amounts of samples. This approach uses model package defined way of saving outputs and then moving all generated batches to output directory.

The one thing I cannot do nicely is adding the sample index to the final filenames. (For cases where models outputs several files for one sample like image and mask)

### CODE REVIEW

File: src/medigan/execute_model/model_executor.py [hunk 1]
```diff
@@ -344,8 +347,30 @@ def gen(**some_other_kwargs):
...
+
+                    batch_path = os.path.join(output_path, "batch_" + str(batch_num))
+                    prepared_kwargs.update({"output_path": batch_path})
+
+                    generate_method(**prepared_kwargs)
+
+                    for filename in os.listdir(batch_path):
+                        os.rename(
+                            os.path.join(batch_path, filename),
+                            os.path.join(output_path, str(batch_num) + "_" + filename),
```
Comment (RichardObi) [inline src/medigan/execute_model/model_executor.py:366]:
```suggestion
                            os.path.join(output_path, "batch_" + str(batch_num) + "_" + filename),
```

File: src/medigan/execute_model/model_executor.py [hunk 2]
```diff
@@ -284,6 +286,7 @@ def generate(
         output_path: str = None,
         save_images: bool = True,
         is_gen_function_returned: bool = False,
+        batch_size: int = 32,
```
Comment (RichardObi) [inline src/medigan/execute_model/model_executor.py:289]: Description needs to be added to the docstring of the method.

**Review by RichardObi — APPROVED** [review]
Great work! I tested this PR locally and everything worked fine and as expected. 👍 
Please check my 2 minor comments before merging.

Regarding your comment, the model packages should handle the cases where models output several files for one sample like image and mask (e.g. resulting in pairs like "batch_i_j_img.png", "batch_i_j_mask.png" with corresponding j)

**Comment by faildeny** [comment]
According to #37, saving samples in such a format (every output type in the same directory) makes it hard to compute fid. Do you think we could split them into separate directory?

**Comment by RichardObi** [comment]
> According to #37, saving samples in such a format (every output type in the same directory) makes it hard to compute fid. Do you think we could split them into separate directory?

Thanks for making me aware of issue #37 and the case for FID. As you also pointed out, the second option would in #37 would make FID (and similar) calculations more straightforward, but, on the other hand, an update of medigan's guidelines and models (e.g., 00004, polyp models) would be needed. To avoid the need to guideline and model updates while still having maximum flexibility of file storage pattern for users, I suggest a hybrid approach: 
An additional parameter of the `generate` function, e.g., `create_output_folder_by_type: bool`. If true, medigan separates the outputs of different types (e.g., masks and imgs) into separate folders as suggested in option 2 in #37. Else, all generated files are in the same folder as in option 1 in #37.

**Comment by faildeny** [comment]
Bugs fixed and all tests pass.

**Pull request merged**